PWA-622: Update Venia's button

davemacaulay opened the pull request:
## Description
Adjust `Button` styles to match the new design

## Related Issue
https://jira.corp.magento.com/browse/PWA-622

## Acceptance 
- Verify all buttons are updated throughout the app
- Ensure there is no broken functionality with the styling change

### Verification Stakeholders
@jimbo 
@dhaecker 
@soumya-ashok 
@schensley 

### Specification
<!-- Changes to `upward-spec` and/or `upward-js` packages must be reviewed -->
<!-- by `UPWARD-PHP` maintainers to ensure continued compatibility -->

### Verification Steps
1. Navigate around the app following all known user flows
2. Verify all buttons previously present are present and work
3. Verify all buttons are positioned and readable

## Screenshots / Screen Captures (if appropriate)

## Checklist
<!--- Go over all the following points, and make sure you've done anything necessary -->
- [x] I have added tests to cover my changes, if necessary.
- [x] I have updated the documentation accordingly, if necessary.

devops-pwa-codebuild commented:
**Performance Test Results**

The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.

https://pr-2496.pwa-venia.com : LH Performance Expected 0.85 Actual 0.57, LH Best Practices Expected 1 Actual 0.92
https://pr-2496.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.35, LH Best Practices Expected 1 Actual 0.92
https://pr-2496.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.5, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92

sirugh commented:
Just a quick glance, I noticed the following:

1) Filter/sort buttons on category page seem to have been missed:

[![Image from Gyazo](https://i.gyazo.com/1d8540269fff919aa545708134917498.png)](https://gyazo.com/1d8540269fff919aa545708134917498)

2) Estimate Shipping on Cart page doesn't match other adjustment button style:

[![Image from Gyazo](https://i.gyazo.com/3020d37b80bbe4f07bd7c536160ccc5d.png)](https://gyazo.com/3020d37b80bbe4f07bd7c536160ccc5d)

3) Checkout button on Cart page has a weird shadow thing:

[![Image from Gyazo](https://i.gyazo.com/b926275998f379159f75dfdc7e714001.gif)](https://gyazo.com/b926275998f379159f75dfdc7e714001)

4) Some icons are blue stroke and some are black stroke:

[![Image from Gyazo](https://i.gyazo.com/9330a57305f525e8b97f62f000fcbb56.gif)](https://gyazo.com/9330a57305f525e8b97f62f000fcbb56)

davemacaulay commented:
> Just a quick glance, I noticed the following:
> 1. Filter/sort buttons on category page seem to have been missed:

Good catch, spoke with Soumya and updated to black outline with smaller min width.

> 1. Estimate Shipping on Cart page doesn't match other adjustment button style:

Another great catch, fixed.

> 1. Checkout button on Cart page (and other buttons elsewhere) has a weird shadow thing:

This is intentional, this is the `:focus` state for buttons and will be displayed in _some_ (notably Chrome) browsers when you click on a button.

> 1. Some icons are blue stroke and some are black stroke:

Didn't know account area was a thing, thanks for bringing it to my attention. Worked with Scott & Soumya and decided to turn the black icon blue.

davemacaulay commented on packages/venia-ui/lib/components/Checkout/checkoutButton.css:
```diff
@@ -1,4 +1,5 @@
 .icon {
     composes: root from '../Icon/icon.css';
     --stroke: white;
+    margin-top: -2px;
```
The icon is very bottom heavy, so moving it up slightly improves the visual appearance of the checkout button.

anthoula commented on packages/venia-ui/lib/components/AuthBar/userChip.css:
```diff
@@ -18,6 +18,11 @@
 
+.avatarIcon {
+    composes: root from '../Icon/icon.css';
+    --stroke: rgb(var(--venia-global-color-teal));
```
I would avoid using teal, and go straight for the blues, so in this case:
`--venia-global-color-blue-400`

jimbo commented on packages/venia-ui/lib/components/Button/button.css:
```diff
@@ -1,72 +1,89 @@
-    outline: none;
-    transition-duration: 128ms;
-    --color: var(--venia-global-color-teal);
+    box-shadow: -6px 6px rgba(var(--venia-global-color-blue-700), 0.3);
```
Now that these property values are space-delimited rather than comma-delimited, we need to use the new syntax in which the alpha channel is separated by a slash rather than a comma. This syntax is supported by both `rgb()` and `rgba()`, so we can use the former.

```suggestion
    box-shadow: -6px 6px rgb(var(--venia-global-color-blue-700) / 0.3);
```

davemacaulay replied:
Updated all instances of `rgba(var` with the new `rob(var(...) /` syntax :)

jimbo commented on packages/venia-ui/lib/components/CartPage/PriceAdjustments/CouponCode/couponCode.css:
```diff
@@ -7,10 +7,7 @@
-
-    /* match the height of the input form. */
-    height: 2.25rem;
+    align-self: flex-end;
```
```suggestion
    align-self: end;
```

jimbo commented on packages/venia-ui/lib/components/CartPage/PriceAdjustments/ShippingMethods/shippingMethods.css:
```diff
@@ -34,6 +26,11 @@
 
+.estimateButton {
+    composes: root_normalPriority from '../../../Button/button.css';
+    justify-self: flex-start;
```
```suggestion
    justify-self: start;
```

dpatil-magento commented:
@davemacaulay  As discussed Shipping Info and Shipping Method Edit overlay buttons needs to be consistent in guest and Auth mode. Rest all looks good.

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js:
```diff
@@ -84,12 +77,8 @@ const CustomerForm = props => {
-            root_highPriority: classes.submit_update
-        },
         disabled: isSaving,
-        priority: isUpdate ? 'high' : 'normal',
```
I doubt this was an accident originally. Does the style guide call for all proceed/submit buttons to be high priority?

davemacaulay replied:
Both 'Update' and 'Add' are considered primary actions in the address modal, I modified this logic to the `Save and Continue` button is the correct `normal` priority if the authenticated user doesn't have any saved addresses :)

dpatil-magento commented:
QA Approved.  
Follow up ticket will handle - 
1. Mobile View >> Filters >> Reset filter does not get disabled after first tap but clears filters (minor)
2. Mini cart Edit product - Cancel button UI and order of display is stale (missed to report/update)
